src/ape/types/__init__.py

```diff
@@ -275,6 +302,21 @@ def __getitem__(self, item: str) -> Any:
     def get(self, item: str, default: Optional[Any] = None) -> Any:
         return self.event_arguments.get(item, default)
 
+class MockContractLog(BaseContractLog):
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, BaseContractLog):
+            return NotImplemented
```

Comment:
I think we want to be stricter to say we can only compare a Mock to a normal `ContractLog`
```suggestion
        if not isinstance(other, ContractLog):
            return NotImplementedError()
```